src/hangar/checkout.py

```diff
@@ -185,11 +187,35 @@ def close(self) -> None:
         multiple simultaneous read checkouts.
         '''
         self.__verify_checkout_alive()
-        self._datasets._close()
+        with suppress(AttributeError):
+            self._datasets._close()
+
+        for dsetn in (self._datasets._datasets.keys()):
+            for attr in list(self._datasets._datasets[dsetn].__dir__()):
+                with suppress(AttributeError, TypeError):
```

Comment:
Minor nit: can't we use a single `with` statement for all three loops?

Reply:
actually no, since if a `with suppress(FooException):` block encounters an Exception, it will exit the `with` block and continue execution at the next line outside the block.